src/shims/windows/sync.rs

```diff
@@ -8,6 +8,38 @@ use crate::*;
 
 const SRWLOCK_ID_OFFSET: u64 = 0;
 const INIT_ONCE_ID_OFFSET: u64 = 0;
+const CONDVAR_ID_OFFSET: u64 = 0;
+
+impl<'mir, 'tcx> EvalContextExtPriv<'mir, 'tcx> for crate::MiriInterpCx<'mir, 'tcx> {}
+pub trait EvalContextExtPriv<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
+    /// Try to reacquire the lock associated with the condition variable after we
+    /// were signaled.
+    fn reacquire_cond_lock(
+        &mut self,
+        thread: ThreadId,
+        lock: RwLockId,
+        shared: bool,
+    ) -> InterpResult<'tcx> {
+        let this = self.eval_context_mut();
+        this.unblock_thread(thread);
+
+        if shared {
+            if this.rwlock_is_locked(lock) {
+                this.rwlock_enqueue_and_block_reader(lock, thread);
+            } else {
+                this.rwlock_reader_lock(lock, thread);
+            }
+        } else {
+            if this.rwlock_is_write_locked(lock) {
```

Comment:
So if it's read locked when we write lock we don't enqueue but immediately do a write lock?

Reply:
Whoops, `rwlock_is_locked` and `rwlock_is_write_locked` are swapped here. It doesn't fail any tests because std only ever uses condvars in exclusive mode. I'll add tests for shared mode.